packages/ui-number-input/src/NumberInput/index.tsx

```diff
@@ -58,207 +61,242 @@ category: components
+          inputRef.current?.focus()
+
+          if (typeof callback === 'function') {
+            callback(event)
+          }
+        }
+      },
+      [interaction]
+    )
 
-      if (typeof callback === 'function') {
-        callback(event)
-      }
-    }
-  }
+    const handleClickUpArrow = useCallback(
+      (event: React.MouseEvent<HTMLButtonElement>) => {
+        arrowClicked(event, onIncrement)
+      },
+      [arrowClicked, onIncrement]
+    )
 
-  renderArrows(customIcons?: { increase: Renderable; decrease: Renderable }) {
-    return (
-      <span css={this.props.styles?.arrowContainer}>
-        <button
-          aria-hidden
-          css={this.props.styles?.arrow}
-          onMouseDown={this.handleClickUpArrow}
-          tabIndex={-1}
-          type="button"
-        >
-          {customIcons?.increase ? (
-            callRenderProp(customIcons.increase)
-          ) : (
-            <IconArrowOpenUpLine />
-          )}
-        </button>
-        <button
-          aria-hidden
-          css={this.props.styles?.arrow}
-          onMouseDown={this.handleClickDownArrow}
-          tabIndex={-1}
-          type="button"
-        >
-          {customIcons?.decrease ? (
-            callRenderProp(customIcons.decrease)
-          ) : (
-            <IconArrowOpenDownLine />
-          )}
-        </button>
-      </span>
+    const handleClickDownArrow = useCallback(
+      (event: React.MouseEvent<HTMLButtonElement>) => {
+        arrowClicked(event, onDecrement)
+      },
+      [arrowClicked, onDecrement]
     )
-  }
 
-  render() {
-    const {
-      renderLabel,
-      display,
-      placeholder,
-      isRequired,
-      showArrows,
-      value,
-      width,
-      styles,
-      allowStringValue,
-      renderIcons,
-      margin
-    } = this.props
+    // Expose imperative API via ref
+    useImperativeHandle(
+      ref,
+      () => ({
+        focus: () => {
+          inputRef.current?.focus()
+        },
+        get focused() {
+          return isActiveElement(inputRef.current)
+        },
+        get id() {
+          return id
+        },
+        get invalid() {
+          return invalid
+        },
+        get interaction() {
+          return interaction
+        },
+        get value() {
+          return inputRef.current?.value
+        }
+      }),
+      [id, invalid, interaction]
+    )
 
-    const { interaction } = this
+    // Render methods
+    const renderArrows = (customIcons?: {
+      increase: Renderable
+      decrease: Renderable
+    }) => {
+      return (
+        <span css={styles?.arrowContainer}>
+          <button
+            aria-hidden
+            css={styles?.arrow}
+            onMouseDown={handleClickUpArrow}
+            tabIndex={-1}
+            type="button"
+          >
+            {customIcons?.increase ? (
+              callRenderProp(customIcons.increase)
+            ) : (
+              <IconArrowOpenUpLine />
```

Comment:
We should define the icon colors here for each state, now it stays black in every theme and state.